File: Bahasa/app/src/main/java/ds/id/Bahasa/Common/KataSetting.java
package ds.id.Bahasa.Common;

import ds.id.Bahasa.Controls.SharedPreferencesUtil;

public class KataSetting {

    private static KataSetting instance;
    public static KataSetting getInstance(){
        if( instance == null){
            instance = new KataSetting();
        }
        return instance;
    }

    //앱 버전
    public void setKataVersion(String sValue){
        SharedPreferencesUtil.getInstance().setValue("KataVersion", sValue);
    }

    public String getKataVersion() {
        String sValue = (String)SharedPreferencesUtil.getInstance().getValue("KataVersion", "");
        return sValue;
    }

    //현재 화면 번호
    public void setKataPlayNumber(String sValue){
        SharedPreferencesUtil.getInstance().setValue("KataPlayNumber", sValue);
    }

    public String getKataPlayNumber() {
        String sValue = (String)SharedPreferencesUtil.getInstance().getValue("KataPlayNumber", "");
        return sValue;
    }

    //자동 반복 시간
    public void setKataTime(int nValue){
        SharedPreferencesUtil.getInstance().setValue("KataTime", nValue);
    }

    public int getKataTime() {
        int nValue = (int)SharedPreferencesUtil.getInstance().getValue("KataTime", 10);
        return nValue;
    }

    public void setWordKataTime(int nValue){
        SharedPreferencesUtil.getInstance().setValue("WordKataTime", nValue);
    }

    public int getWordKataTime() {
        int nValue = (int)SharedPreferencesUtil.getInstance().getValue("WordKataTime", 10);
        return nValue;
    }

    //문장 사운드 지원
    public void setKalimatSound(Boolean bValue){
        SharedPreferencesUtil.getInstance().setValue("KalimatSound", bValue);
    }

    public Boolean getKalimatSound() {
        Boolean bValue = (Boolean)SharedPreferencesUtil.getInstance().getValue("KalimatSound", false);
        return bValue;
    }

    //단어장 사운드 지원
    public void setKataSound(Boolean bValue){
        SharedPreferencesUtil.getInstance().setValue("kataSound", bValue);
    }

    public Boolean getKataSound() {
        Boolean bValue = (Boolean)SharedPreferencesUtil.getInstance().getValue("kataSound", false);
        return bValue;
    }

    //스크린 화면 항상켜기
    public void setScreenLock(Boolean bValue){
        SharedPreferencesUtil.getInstance().setValue("ScreenLock", bValue);
    }

    public Boolean getScreenLock() {
        Boolean bValue = (Boolean)SharedPreferencesUtil.getInstance().getValue("ScreenLock", false);
        return bValue;
    }

    //광고 하루에 한번만
    public void setUpDateTime(String sValue){
        SharedPreferencesUtil.getInstance().setValue("AdvUpdateTime", sValue);
    }

    public String getUpDateTime() {
        String sValue = (String)SharedPreferencesUtil.getInstance().getValue("AdvUpdateTime", "");
        return sValue;
    }

    //설명문 한번만
    public void setIntroPopup(Boolean bValue){
        SharedPreferencesUtil.getInstance().setValue("Intro", bValue);
    }

    public Boolean getIntroPopup() {
        Boolean bValue = (Boolean)SharedPreferencesUtil.getInstance().getValue("Intro", false);
        return bValue;
    }

    //스킨 타입
    public void setSkinStyle(int nValue){
        SharedPreferencesUtil.getInstance().setValue("SkinStyle", nValue);
    }

    public int getSkinStyle() {
        int nValue = (int)SharedPreferencesUtil.getInstance().getValue("SkinStyle", 0);
        return nValue;
    }

    //자동 반복 횟수
    public void setKataTimeRepeat(int nValue){
        SharedPreferencesUtil.getInstance().setValue("KataTimeRepeat", nValue);
    }

    public int getKataTimeRepeat() {
        int nValue = (int)SharedPreferencesUtil.getInstance().getValue("KataTimeRepeat", 0);
        return nValue;
    }

    public void setWordKataTimeRepeat(int nValue){
        SharedPreferencesUtil.getInstance().setValue("WordKataTimeRepeat", nValue);
    }

    public int getWordKataTimeRepeat() {
        int nValue = (int)SharedPreferencesUtil.getInstance().getValue("WordKataTimeRepeat", 10);
        return nValue;
    }

    public void deleteKataSetting(String sKey){
        SharedPreferencesUtil.getInstance().remove(sKey);
    }
}

File: Bahasa(kotlin)/app/src/main/java/ds/id/Bahasa/Common/KataSetting.kt
package ds.id.Bahasa.Common

import ds.id.Bahasa.Controls.SharedPreferencesUtil

class KataSetting {

    private val tag: String = KataSetting::class.java.simpleName

    companion object {
        private var instance: KataSetting? = null
        fun getInstance(): KataSetting {
            if (instance == null) {
                instance = KataSetting()
            }
            return instance as KataSetting
        }
    }

    //앱 버전
    var kataVersion: String?
        get() = SharedPreferencesUtil.getInstance().getValue("KataVersion", "") as String?
        set(sValue) {
            SharedPreferencesUtil.getInstance().setValue("KataVersion", sValue!!)
        }

    //현재 화면 번호
    var kataPlayNumber: String?
        get() = SharedPreferencesUtil.getInstance().getValue("KataPlayNumber", "") as String?
        set(sValue) {
            SharedPreferencesUtil.getInstance().setValue("KataPlayNumber", sValue!!)
        }

    //자동 반복 시간
    var kataTime: Int
        get() = SharedPreferencesUtil.getInstance().getValue("KataTime", 10) as Int
        set(nValue) {
            SharedPreferencesUtil.getInstance().setValue("KataTime", nValue)
        }

    var wordKataTime: Int
        get() = SharedPreferencesUtil.getInstance().getValue("WordKataTime", 10) as Int
        set(nValue) {
            SharedPreferencesUtil.getInstance().setValue("WordKataTime", nValue)
        }

    //문장 사운드 지원
    var kalimatSound: Boolean?
        get() = SharedPreferencesUtil.getInstance().getValue("KalimatSound", false) as Boolean?
        set(bValue) {
            SharedPreferencesUtil.getInstance().setValue("KalimatSound", bValue!!)
        }

    //단어장 사운드 지원
    var kataSound: Boolean?
        get() = SharedPreferencesUtil.getInstance().getValue("kataSound", false) as Boolean?
        set(bValue) {
            SharedPreferencesUtil.getInstance().setValue("kataSound", bValue!!)
        }

    //스크린 화면 항상켜기
    var screenLock: Boolean?
        get() = SharedPreferencesUtil.getInstance().getValue("ScreenLock", false) as Boolean?
        set(bValue) {
            SharedPreferencesUtil.getInstance().setValue("ScreenLock", bValue!!)
        }

    //광고 하루에 한번만
    var upDateTime: String?
        get() = SharedPreferencesUtil.getInstance().getValue("AdvUpdateTime", "") as String?
        set(sValue) {
            SharedPreferencesUtil.getInstance().setValue("AdvUpdateTime", sValue!!)
        }

    //설명문 한번만
    var introPopup: Boolean?
        get() = SharedPreferencesUtil.getInstance().getValue("Intro", false) as Boolean?
        set(bValue) {
            SharedPreferencesUtil.getInstance().setValue("Intro", bValue!!)
        }

    //스킨 타입
    var skinStyle: Int
        get() = SharedPreferencesUtil.getInstance().getValue("SkinStyle", 0) as Int
        set(nValue) {
            SharedPreferencesUtil.getInstance().setValue("SkinStyle", nValue)
        }

    //자동 반복 횟수
    var kataTimeRepeat: Int
        get() = SharedPreferencesUtil.getInstance().getValue("KataTimeRepeat", 0) as Int
        set(nValue) {
            SharedPreferencesUtil.getInstance().setValue("KataTimeRepeat", nValue)
        }

    var wordKataTimeRepeat: Int
        get() = SharedPreferencesUtil.getInstance().getValue("WordKataTimeRepeat", 10) as Int
        set(nValue) {
            SharedPreferencesUtil.getInstance().setValue("WordKataTimeRepeat", nValue)
        }

    fun deleteKataSetting(sKey: String?) {
        SharedPreferencesUtil.getInstance().remove(sKey)
    }
}
